app/client/components/LinkingState.ts

```diff
@@ -194,13 +200,111 @@ export class LinkingState extends Disposable {
+      //
+      // However, each linkingState needs to decide whether to use the cursorPos from the srcSec (i.e. its activeRowId),
+      // or to use the previous linkState's incomingCursorPos. We want to use whichever section the user most recently
+      // interacted with, i.e. whichever cursor update was most recent. For this we use, the cursor version (given in
+      // viewSection.lastCursorEdit). incomingCursorPos is a pair of [rowId, sequenceNum], so each linkingState sets its
+      // incomingCursorPos to whichever is most recent between its srcSection, and the previous LS's incCursPos.
+      //
+      // The end result is that there is that because the lastCursorEdits are guaranteed to be unique, there is always
+      // a stable configuration of links, where even in the case of a cycle the incomingCursorPos will all take their
+      // rowId and version from the most recently edited viewSection in the cycle, which is what the user expects
+      //
+      //               ...from C--> [A.LS] -------->  [B.LS]               --> [C.LS] ----->...to A                    |
+      //                               |                 |                /       |                                    |
+      //                               v                 v               /        v                                    |
+      //                             [ A ]             [ B ]   ---------/       [ C ]                                  |
+      //                                          (most recently edited)                                               |
+      //
+      // Once the incomingCursorPos-es are determined correctly, the cursorPos-es just need to pull out the rowId,
+      // and that will drive the cursors of the associated tgt section for each LS.
+      //
+      // NOTE: setting cursorPos will change the viewSections' cursor, but it's special-cased to
+      // not affect their lastCursorEdit times.
+      // =============================
+
+      // gets the relevant col value for the passed-in rowId, or return rowId unchanged if same-table link
       const srcValueFunc = this._makeValGetter(this._srcSection.table(), this._srcColId);
 
-      if (srcValueFunc) { // if makeValGetter succeeded, set up cursorPos
-        this.cursorPos = this.autoDispose(ko.computed(() =>
-          srcValueFunc(srcSection.activeRowId()) as UIRowId
-        ));
+      // check for failure
+      if (srcValueFunc) {
+        //Incoming-cursor-pos determines what the linked cursor position should be, considering the previous
+        //linked section (srcSection) and all upstream sections (through srcSection.linkingState)
+        this.incomingCursorPos = this.autoDispose((ko.computed(() => {
+          // Get previous linkingstate's info
+          // NOTE: prevlink = this.srcSec.linkingState is 2 hops back, (i.e. it reads from from this.srcSec.linkSrcSec)
```

Comment:
```suggestion
          // NOTE: prevlink = this.srcSec.linkingState is 2 hops back, (i.e. it reads from this.srcSec.linkSrcSec)
```

Can you rewrite it and add more explanation I don't follow this..